src/generation_tools.jl

```diff
@@ -15,16 +14,15 @@ Arguments:
 """
 ϕ_tuple(t::NTuple) = atan(t[2],t[1])
 
-
-
-
 ## conversion_tools.jl
 
 """
     idx_to_dim(idx_arr) 
 
 Converts an N-dimensional array of NTuple to an N+1 dimensional array by orienting the (inner) tuple along the (outer) trailing+1 dimension.
+Note that this function is not type stable!
```

Comment:
Can you double check if this is really not type stable? 
If it does what you say, it should

Reply:
I think it is type stable.
The size parameter is obtained from the Type of the Tuple, which is part of the array type.
Do you think otherwise?